src/app/shared/modules/header/header.component.ts

```diff
@@ -97,6 +99,17 @@ export class HeaderComponent implements OnInit, OnDestroy {
     }
   }
 
+  createButtonShouldBeDisplayed(path: string): boolean {
+    switch (path) {
```

Comment:
To complicated. Please simplify, e.g.

```
createButtonShouldBeDisplayed(path: string): boolean {
   return path === '/products'
 }
```

and once you simplify it as here - you will see that this method is not needed at all. 

just move the body to the plate where you were using this method.

`this.displayCreateButton = event.url === '/products'`

Reply:
Ok, using just `this.displayCreateButton = event.url === '/products'` didn't work exactly as intended because it was not showing the button when the "/" address was used and redirected to products so I made it `this.displayCreateButton = event.url === '/products' || event.url === '/'`.